metrics_proto_api_test.go

```diff
@@ -0,0 +1,1141 @@
+
+	// Adjust description to original description of summary metris.
+	outMDSummaryCount.Description = inMDSummary.Description
+	outMDSummarySum.Description = inMDSummary.Description
+	outMDSummaryPercentile.Description = inMDSummary.Description
+	outMDSummaryCount.Unit = "1"
+	lbl := &labelpb.LabelDescriptor{
+		Key:         "percentile",
+		ValueType:   labelpb.LabelDescriptor_STRING,
+		Description: "the value at a given percentile of a distribution",
+	}
+	outMDSummaryPercentile.Labels = append(outMDSummaryPercentile.Labels, lbl)
+
+	inPointsSummary := createMetricPbPointSummary(10, 119.0, inPercentile, 5.6, 9.6, 12.6, 17.6)
+	outPointSummaryCount := createGoogleMetricPbPointInt64(10)
+	outPointSummarySum := createGoogleMetricPbPointDouble(119.0, true)
+	outPointSummaryPercentile1 := createGoogleMetricPbPointDouble(5.6, false)
+	outPointSummaryPercentile2 := createGoogleMetricPbPointDouble(9.6, false)
+	outPointSummaryPercentile3 := createGoogleMetricPbPointDouble(12.6, false)
+	outPointSummaryPercentile4 := createGoogleMetricPbPointDouble(17.6, false)
+
+	tcs := []*testCases{
+		{
+			name: "Metrics of different types",
+			inMetric: []*metricspb.Metric{
+				{
+					MetricDescriptor: inMDCummDouble,
+					Timeseries: []*metricspb.TimeSeries{
+						{
+							StartTimestamp: startTimestamp,
+							LabelValues:    []*metricspb.LabelValue{inEmptyValue, inOperTypeValue1},
+							Points:         inPointsFloat64_1,
+						},
+					},
+				},
+				{
+					MetricDescriptor: inMDGaugeDouble,
+					Timeseries: []*metricspb.TimeSeries{
+						{
+							LabelValues: []*metricspb.LabelValue{inEmptyValue, inOperTypeValue2},
+							Points:      inPointsFloat64_1,
+						},
+					},
+				},
+				{
+					MetricDescriptor: inMDCummInt64,
+					Timeseries: []*metricspb.TimeSeries{
+						{
+							StartTimestamp: startTimestamp,
+							LabelValues:    []*metricspb.LabelValue{inEmptyValue, inOperTypeValue1},
+							Points:         inPointsInt64_1,
+						},
+					},
+				},
+				{
+					MetricDescriptor: inMDGaugeInt64,
+					Timeseries: []*metricspb.TimeSeries{
+						{
+							LabelValues: []*metricspb.LabelValue{inEmptyValue, inOperTypeValue2},
+							Points:      inPointsInt64_1,
+						},
+					},
+				},
+				{
+					MetricDescriptor: inMDCummDist,
+					Timeseries: []*metricspb.TimeSeries{
+						{
+							StartTimestamp: startTimestamp,
+							LabelValues:    []*metricspb.LabelValue{inEmptyValue, inOperTypeValue1},
+							Points:         []*metricspb.Point{inPointsCummDist},
+						},
+					},
+				},
+				{
+					MetricDescriptor: inMDGuageDist,
+					Timeseries: []*metricspb.TimeSeries{
+						{
+							StartTimestamp: startTimestamp,
+							LabelValues:    []*metricspb.LabelValue{inEmptyValue, inOperTypeValue1},
+							Points:         []*metricspb.Point{inPointsGaugeDist},
+						},
+					},
+				},
+				{
+					MetricDescriptor: inMDSummary,
+					Timeseries: []*metricspb.TimeSeries{
```

Comment:
Consider adding tests on one metric with more than one time series. That's how we found #214.

Reply:
#214 is caught by the summary test. Summary test here translates into 4 guage timeseries (one for each percentile).  In fact, the summary test failed until I rebased the fix #214.

Reply:
I take it back. #214 requires additional test as you suggested. What I had encountered was #209. I'll add the test in another PR.